### PR TITLE
Groovy

### PR DESCRIPTION
**Pull request by mickeyxu**
Added gyro drift correction part from Madgwick's report for the MARG implementation.

### CODE REVIEW

**Comment by mintar** [comment]
Looks good to me. As far as I can see, this shouldn't break any old configs, because the default value for zeta is 0. Right?

**Comment by mickeyxu** [comment]
That's right!

**Comment by lorenzoriano** [comment]
I can't access the hardware this week, but it looks good to me and if you guys (@mintar ?) agree we can merge the branch. We should also make it publicly available in the ROS repositories.

File: imu_filter_madgwick/src/imu_filter.cpp [hunk 1]
```diff
@@ -305,11 +307,36 @@ void ImuFilter::madgwickAHRSupdate(
 		s2 *= recipNorm;
 		s3 *= recipNorm;
 
+		// compute gyro drift bias
+		_w_err_x = _2q0 * s1 - _2q1 * s0 - _2q2 * s3 + _2q3 * s2;
+		_w_err_y = _2q0 * s2 + _2q1 * s3 - _2q2 * s0 - _2q3 * s1;
+		_w_err_z = _2q0 * s3 - _2q1 * s2 + _2q2 * s1 - _2q3 * s0;
+		
+		w_bx_ += _w_err_x * dt * zeta;
```
Comment (lorenzoriano) [inline imu_filter_madgwick/src/imu_filter.cpp:315]: this should be zeta_

**Comment by lorenzoriano** [comment]
See the above comment, there's a syntax error

**Comment by mickeyxu** [comment]
my bad...

Can anyone please fix that directly in the code?

**Comment by mintar** [comment]
Ouch. And I was 99% sure that I had tested this before merging. Anyways, it's fixed in 7257d1d07fed4d0cb5e280e9cee3f31b7ce40937 (for fuerte) / 2d9b3e1df2c2848a3fbdf6251e9150ef704f45d5 (for groovy). Good catch, @lorenzoriano ! :-)

**Pull request merged**